packages/blaze-query/src/blockfrost.ts

```diff
@@ -138,86 +143,297 @@ export class Blockfrost implements Provider {
+      const query = `/txs/${txIn.transactionId()}/utxos`;
+      const json = await fetch(`${this.url}${query}`, {
+        headers: this.headers(),
+      }).then((resp) => resp.json());
+
+      if (!json) {
+        throw new Error("resolveUnspentOutputs: Could not parse response json");
+      }
+
+      const response =
+        json as BlockfrostResponse<BlockfrostUnspentOutputResolution>;
+
+      if ("message" in response) {
+        throw new Error(
+          `resolveUnspentOutputs: Blockfrost threw "${response.message}"`,
+        );
+      }
+
+      const txIndex = BigInt(txIn.index());
+
+      for (const blockfrostUTxO of response.outputs) {
+        if (BigInt(blockfrostUTxO.output_index) !== txIndex) {
+          // Ignore outputs whose index don't match
+          // the index we are looking for
+          continue;
+        }
+        // Blockfrost API does not return tx hash, so it must be manually set
+        blockfrostUTxO.tx_hash = txIn.transactionId();
+
+        const buildTxUnspentOutput = buildTransactionUnspentOutput(
+          Address.fromBech32(blockfrostUTxO.address),
+        );
+
+        results.add(buildTxUnspentOutput(blockfrostUTxO));
+      }
+    }
+
+    return Array.from(results);
   }
 
-  resolveDatum(_datumHash: DatumHash): Promise<PlutusData> {
-    throw new Error("unimplemented");
+  /**
+   * This method returns the datum for the datum hash given as argument.
+   * The response is parsed into a PlutusData type, which is then returned.
+   * If the response is not in the expected format, an error is thrown.
+   * @param datumHash - The hash of a datum
+   * @returns A Promise that resolves to PlutusData
+   */
+  async resolveDatum(datumHash: DatumHash): Promise<PlutusData> {
+    const query = `/scripts/datum/${datumHash}/cbor`;
+    const json = await fetch(`${this.url}${query}`, {
+      headers: this.headers(),
+    }).then((resp) => resp.json());
+
+    if (!json) {
+      throw new Error("resolveDatum: Could not parse response json");
+    }
+
+    const response = json as BlockfrostResponse<BlockfrostDatumHashResolution>;
+
+    if ("message" in response) {
+      throw new Error(`resolveDatum: Blockfrost threw "${response.message}"`);
+    }
+
+    return PlutusData.fromCbor(HexBlob(response.cbor));
   }
 
+  /**
+   * This method awaits confirmation of the transaction given as argument.
+   * The response is parsed into a boolean, which is then returned.
+   * If tx is not confirmed at first and no value for timeout is provided,
+   * then false is returned.
+   * If tx is not confirmed at first and a value for timeout (in ms) is given,
+   * then subsequent checks will be performed at a 20 second interval until
+   * timeout is reached.
+   * @param txId - The hash of a transaction
+   * @param timeout - An optional timeout for waiting for confirmation. This
+   * value should be greater than average block time of 20000 ms
+   * @returns A Promise that resolves to a boolean
+   */
```

Comment:
@micahkendall I suspect the current implementation of this method on the Maestro provider is incorrect. I wrote down what I assume to be the expected behavior as comments above and took the liberty to change the implementation to match my assumptions.